packages/postgres-database/src/simcore_postgres_database/migration/versions/28aa90807453_product_name_column_in_api_keys_table.py

```diff
@@ -0,0 +1,51 @@
+"""product_name column in api_keys table
+
+Revision ID: 28aa90807453
+Revises: be0dece4e67c
+Create Date: 2023-10-24 14:54:38.856859+00:00
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "28aa90807453"
+down_revision = "be0dece4e67c"
+branch_labels = None
+depends_on = None
+
+
+def find_default_product_name_or_none(conn):
+    query = sa.text("SELECT name FROM products ORDER BY priority LIMIT 1")
+    result = conn.execute(query)
+    row = result.fetchone()
+    return row[0] if row else None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column("api_keys", sa.Column("product_name", sa.String(), nullable=False))
+    op.create_foreign_key(
+        "fk_api_keys_product_name",
+        "api_keys",
+        "products",
+        ["product_name"],
+        ["name"],
+        onupdate="CASCADE",
+        ondelete="CASCADE",
+    )
+    # ### end Alembic commands ###
+
+    conn = op.get_bind()
+
+    default_product = find_default_product_name_or_none(conn)
```

Comment:
Yes, this is one of the options for how to do the migration, if we are okay with that we will bound the API keys to the default product. Which I guess is almost everywhere `osparc` ? Another more complicated option is we duplicate these keys for each product the user has access to, but if it is not necessary we do not need to do it.

Reply:
Yes only that produce so far uses the api. In any case it would be good to review it once it moves to master deploy